environments/dotnet20/README.md

```diff
@@ -304,3 +268,98 @@ $ curl -XPOST http://localhost:8888/specialize
 ```
 $ curl -XGET http://localhost:8888
 ``` 
+
+## Few Aditional Features  
+
+**1. NameSpace support :**
+
+ Now , You can use namespace for Fission function class and have many other classes 
+				 in same namespace , this is also backward compatible , however main execution class
+				 would always be *FissionFunction* and its method *Execute*
+
+```
```

Comment:
Would you mind to fix the indent in README (also in builder README.md)?

Reply:
sure i will change indent.

for compilation error , did we put **CsvHelper** in nuget.txt ?
as in previous redme.md that is not highlighted well , and eve i missed to put its refrence and got same error.


Please have a look at this text in readme :
------------------------------------------------------------
Here is my nuget.txt

CsvHelper
As we dont want to exclude any specific dll thus we shall leave exclude.txt as empty
------------------------------------------------------------------------

Reply:
updated readme by highlight this little more, to improve readability .

Reply:
Yes, I put both nuget.txt and func.cs into the zip file. Here is the one I tested with https://github.com/fission/fission/files/3048196/dotnet.zip

Reply:
Hi ,
the package which you shared , there  func.cs file was not having value initilized for string variable **respo** (that was issue in readme which i updated 3 month back's , so taking sample example reference from readme.md , here is the example ):

```
using System;
using Fission.DotNetCore.Api;

public class FissionFunction
{
    public string Execute(FissionContext context){
		string respo = "initial";
	        try
            {
				context.Logger.WriteInfo("Staring..... ");
				respo=$" sample object by getting Enum of   CsvHelper nuget dll: { CsvHelper.Caches.NamedIndex.ToString()}";
            }
            catch(Exception ex)
            {
				context.Logger.WriteError(ex.Message);
                respo = ex.Message;
            }
		context.Logger.WriteInfo("Done!");
		return respo;
    }
}
```

and nuget.txt is correct ,
thus after updating that correct C# function example in your zip ,i could successfully build the package :
Please use this zip :https://github.com/paraspatidar/SampleProjetcs/raw/master/func.zip

---------------------
```
Name:        func-zip-omoc
Environment: dotnetcor1
Status:      succeeded
Build Logs:
Builder Task Begins!
going to create logger!!
detailed logs for this build will be at : logs/2019_05_10_dafae657-3498-48a7-a0bb-ff6e1fd73017.log!!
Latest Package form Exact Search : CsvHelper| 12.1.2 in Repo Nugetdownload
---------------------------------------------------------------------
Latest Package form Exact Search : System.Reflection.TypeExtensions| 4.4.0 in Repo Nugetdownload
---------------------------------------------------------------------
Latest Package form Exact Search : Microsoft.CSharp| 4.4.0 in Repo Nugetdownload
---------------------------------------------------------------------
Package System.Reflection.TypeExtensions is Already Installed at  | /app/Nugetdownload/System.Reflection.TypeExtensions.4.4.0  | skipping instalation !!
Package Microsoft.CSharp is Already Installed at  | /app/Nugetdownload/Microsoft.CSharp.4.4.0  | skipping instalation !!
/app/Nugetdownload/System.Reflection.TypeExtensions.4.4.0/System.Reflection.TypeExtensions.4.4.0.nupkg
/app/Nugetdownload/Microsoft.CSharp.4.4.0/Microsoft.CSharp.4.4.0.nupkg
done for package Microsoft.CSharp , with total Dlls 0
done for package System.Reflection.TypeExtensions , with total Dlls 0
Package CsvHelper is Already Installed at  | /app/Nugetdownload/CsvHelper.12.1.2  | skipping instalation !!
/app/Nugetdownload/CsvHelper.12.1.2/CsvHelper.12.1.2.nupkg
dll Path: /app/Nugetdownload/CsvHelper.12.1.2/lib/netstandard2.0/CsvHelper.dll
done for package CsvHelper , with total Dlls 1
Total Dlls 1 for rootpackage CsvHelper-
done with nuget engine!!!!
DLL Info Gathered!!
Trying to compile it during build itslef !!
Compile Success!!
Compilation result Gathered as : True!!
Copy to Source Done!!
BuildSpecs Done!!
Builder Task Ends!
+ cd /packages/func-zip-omoc-uykmnm
+ echo src : /packages/func-zip-omoc-uykmnm ,dest: /packages/func-zip-omoc-uykmnm-llndku
+ cd /app
+ dotnet Builder.dll
+ cp -r /packages/func-zip-omoc-uykmnm /packages/func-zip-omoc-uykmnm-llndku
---------------------------------
 ```


Also , feel free to let me know for a session , if your would like to do this along side me,

Reply:
After testing, found out that the problem that caused compilation error is that `nuget.txt` didn't allow any blank line in the file like following.
```
CsvHelper:12.1.2

```
which might be too sensitive for the user.

So we have some ways to address this problem when parsing `NugetPackageRegEx` & `ExcludeDllRegEx`:
1. Change regex in builderSetting.json
2. Read line by line and check whether it's a blank line

What do you think about it?

(btw, you may also want to fix indent of sample code in README. I pretty like [Code](https://code.visualstudio.com/) to edit markdown files.)

Reply:
Thats the reason we mentioned about that regex to the user and also mentioned to leave a enter for last line.
As I remember i was facing some challenges last time for this newline thing in regex but I would try again to change that logic to read line by line to prevent user error.

In the meantime can you confirm about the functionality for builder ia as expected and can help fission 2.0 to have nugets on runtime and if we are good on that part?

Reply:
Regex is good. But when it becomes complex, then it's hard for others to change it later. Also, as a tutorial doc, it's hard for every user to understand the regex in tutorial doc (though it's still helpful). Since the dependencies file is small, I think we can spend some time to parse them. Could [IsNullOrWhiteSpace](https://stackoverflow.com/a/6480081) gives any help in this case?

> In the meantime can you confirm about the functionality for builder ia as expected and can help fission 2.0 to have nugets on runtime and if we are good on that part?

Do you mean we keep the compilation at function cold start time? I'm ok to merge this PR first and creating an issue to track for dotnet builder enhancement. What do you think?

Reply:
@life1347  , Made necessary changes for regex and blank line check .
Updated readme example with indent and new regex in builder readme.

now user doesnt need to bother about that new line and blank line for nugext and excludedll files.

Of course you can merge PR and we can then look forward for various option for cold start and compilation as there are many ways :
1. for example as builder has already tried compilation then we can improve environment cold start by saving that precompiled binary in deployment package and use it as it is in function thus saving recompilation at fission environment level.
2. many other possible option to improve cold start and compilation.


Lets see what users also face and suggest and we can track them as improvement item via github issues.